modules/javafx.graphics/src/main/java/com/sun/glass/utils/NativeLibLoader.java

```diff
@@ -301,7 +303,22 @@ private static String cacheLibrary(InputStream is, String name, Class caller) th
         }
         if (write) {
             Path path = f.toPath();
-            Files.copy(is, path);
+            File lockFile = new File(cacheDir, ".lock");
+            try (RandomAccessFile randomAccessFile = new RandomAccessFile(lockFile, "rw");
+                 FileChannel fileChannel = randomAccessFile.getChannel();
+                 FileLock fileLock = fileChannel.lock()) {
+                try {
+                    if (!Files.exists(path)) {
+                        Files.copy(is, path);
```

Comment:
I noticed `is` is not closed (also in the original code).
Would that be a problem?

Reply:
Probably something to evaluate, but in a separate ticket

Reply:
I agree, this should probably be considered as well, as part of a separate ticket. 

Note that, when the file already exists in the cache, the initial InputStream object is used by the DigestInputStream, and it is manually closed and reread again, so special care should be taken here.

Reply:
Unless we suspect a problem, I don't see a need to do anything differently.

Reply:
A possible problem is we might reach the maximum of open file handles in the underlying OS (I've experienced this before).  I think we should definitely fix it (in a separate ticket).

Curiously, the code example for Files.copy(InputStream in, Path target, CopyOption... options) does not include closing of input stream, though it has some verbiage about promptly closing the input stream in case of an error.    I think javadoc there might need to be improved.

Reply:
I've filed https://bugs.openjdk.org/browse/JDK-8314500 for closing the inputStream.